api/src/main/java/io/smallrye/reactive/messaging/ChannelRegistry.java

```diff
@@ -17,10 +17,14 @@
 
     void register(String name, Emitter<?> emitter);
 
+    void register(String name, MutinyEmitter<?> emitter);
+
     List<PublisherBuilder<? extends Message<?>>> getPublishers(String name);
 
     Emitter<?> getEmitter(String name);
 
+    MutinyEmitter<?> getMutinyEmitter(String name);
```

Comment:
Do we have to add this method or "translate" a _regular_ emitter to a MutinyEmitter in the injection point?

Reply:
In theory we could do it all through `getEmitter`, but if I recall it would require hanging onto injection point type in the map of extension to know which is which

Reply:
yeah, so forget about it. Your way is simpler.